src/machine.rs

```diff
@@ -263,12 +263,12 @@ impl MemoryExtra {
                                 symbol_name
                             )
                         });
-                    let (provenance, offset) =
-                        this.memory.create_fn_alloc(FnVal::Other(dlsym)).into_parts();
-                    let ptr = Pointer::new(Some(provenance), offset);
+                    let ptr = this.memory.create_fn_alloc(FnVal::Other(dlsym));
+
                     let place = this.allocate(layout, MiriMemoryKind::ExternStatic.into())?;
                     this.write_pointer(ptr, &place.into())?;
-                    Self::add_extern_static(this, symbol_name, place.ptr);
+
+                    Self::add_extern_static(this, symbol_name, place.ptr.into());
```

Comment:
Why would this branch need `place.ptr.into()` when the other match arms don't?

Reply:
Looks like it was a refactoring error. When I originally wrote the code I used the `ptr` returned from `create_fn_alloc()` which was functionally incorrect and the wrong type. This isn't needed anymore.